drivers/ethernet/dwc_xgmac/Kconfig

```diff
@@ -0,0 +1,78 @@
+#
+# Synopsys DesignWare XGMAC configuration options
+#
+# Copyright(c) 2023, Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+DT_COMPAT_DWC_XGMAC := snps,dwcxgmac
+
+menu "DWC XGMAC configuration"
+
+config ETH_DWC_XGMAC
+	bool "Synopsys DesignWareCore XGMAC Ethernet driver"
+	default y if DT_HAS_SNPS_DWCXGMAC_ENABLED
+	help
+	  This is a driver for the Synopsys DesignWare XGMAC, also referred to
+	  as "DesignWare Cores Ethernet XGMAC". Hardware versions
+	  3.x are supported.
+
+	  Platform specific glue support is also required. This driver is
+	  currently available on targets using the Intel Agilex5 series.
+
+if  ETH_DWC_XGMAC
+
+config ETH_DWC_XGMAC_BOTTOM_HALF_WORK_QUEUE
+	bool "IRQ Bottom Half work Queue"
+	default y
+
+config ETH_DWC_XGMAC_POLLING_MODE
+	bool "Polling mode"
+
+config ETH_DWC_XGMAC_INTERRUPT_POLLING_INTERVAL_US
+	int "XGMAC interrupt polling interval in us"
+	default 500
+	help
+	  XGMAC interrupt polling interval in us
+
+config ETH_XGMAC_MAX_QUEUES
+	int
+	default 8
+	help
+	  Number hardware queues supported by the XGMAC IP
+
+config ETH_DWC_XGMAC_RX_CS_OFFLOAD
+	bool "RX checksum offload"
+	default y
+	help
+	  Enable receive checksum offload for TCP/UDP/IP packets
+
+config ETH_DWC_XGMAC_TX_CS_OFFLOAD
+	bool "TX checksum offload"
+	default y
+	help
+	  Enable transmit checksum offload for TCP/UDP/IP packets
+
+config ETH_DWC_XGMAC_PROMISCUOUS
```

Comment:
CONFIG_NET_PROMISCUOUS_MODE is the standard configuration element that activates promiscuous mode for the entire network subsystem, affecting all network drivers. Consequently, I introduced a driver-specific configuration element to control the activation or deactivation of promiscuous mode specifically for the XGMAC driver.

Reply:
so what is the effect this has? the idea is just the network interface associated with this ethernet will be promiscuous?

Reply:
The purpose is to set this Ethernet interface to a non-promiscuous mode, while the rest of the network stack remains in promiscuous mode. If the intention is to allow all incoming packets on this interface, both CONFIG_NET_PROMISCUOUS_MODE and CONFIG_ETH_DWC_XGMAC_PROMISCUOUS need to be enabled. CONFIG_ETH_DWC_XGMAC_PROMISCUOUS specifically activates hardware-supported promiscuous mode.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/94606265/b876a1c7-b7d3-4ae2-aba8-05ca3736689f)

Reply:
okay, so I assumed the opposite usage of what you had in mind, and what you are saying makes much more sense, as in disabling the promiscuity of one interface instead of the reverse.... interesting, thanks for the response

Reply:
Certainly! Agreeing with you, CONFIG_ETH_DWC_XGMAC_PROMISCUOUS_EXCEPTION seems more fitting. I have updated this name along with making other relevant changes associated with this update. Thanks :-) @decsny